src/tests/ftest/util/server_utils.py

```diff
@@ -49,6 +53,189 @@ class ServerFailed(Exception):
     """Server didn't start/stop properly."""
 
 
+class ServerCommand(CommandWithParameters):
+    """Defines a object representing a server command."""
+
+    def __init__(self, hosts, path):
+        """Create a server Command object"""
+        super(ServerCommand, self).__init__(path, "daos_server")
+
+        self.hosts = hosts
+        self.process = None
+        self.hostfile = None
+
+        self.request = BasicParameter("{}")
+        self.action = BasicParameter("{}")
+        self.targets = FormattedParameter("-t {}")
+        self.config = FormattedParameter("-o {}")
+        self.port = FormattedParameter("-p {}")
+        self.storage = FormattedParameter("-s {}")
+        self.modules = FormattedParameter("-m {}")
+        self.xshelpernr = FormattedParameter("-x {}")
+        self.firstcore = FormattedParameter("-f {}")
+        self.group = FormattedParameter("-g {}")
+        self.attach = FormattedParameter("-a {}")
+        self.sock_dir = FormattedParameter("-d {}")
+        self.insecure = FormattedParameter("-i", None)
+
+    def prepare(self, path, slots):
+        """Prepare the hosts before starting daos server.
+
+        Args:
+            path (str): location to write the hostfile
+            slots (int): slots per host to use in the hostfile
+
+        Raises:
+            ServerFailed: if there is any errors preparing the hosts
+
+        """
+        # Kill any doas servers running on the hosts
+        kill_server(self.hosts)
+
+        # Clean up any files that exist on the hosts
+        clean_server(self.hosts)
+
+        # Ensure the environment for the daos server on each host
+        run_agent(path, self.hosts)
+
+        # Create the hostfile
+        self.hostfile = write_host_file(self.hosts, path, slots)
+
+    def update_configuration(self, basepath):
+        """Update the config parameter with a yaml file.
+
+        Args:
+            basepath (str): DAOS install basepath
+        """
+        self.config.update(create_server_yaml(basepath), "server.config")
+
+    def get_param_names(self):
+        """Get a sorted list of daos_server command parameter names."""
+        names = self.get_attribute_names(FormattedParameter)
+        names.extend(["request", "action"])
+        return names
+
+    def get_params(self, test, path="/run/daos_server/*"):
+        """Get values for all of the server command params using a yaml file.
+
+        Sets each BasicParameter object's value to the yaml key that matches
+        the assigned name of the BasicParameter object in this class. For
+        example, the self.block_size.value will be set to the value in the yaml
+        file with the key 'block_size'.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace.
+                Defaults to "/run/daos_server/*".
+
+        """
+        super(ServerCommand, self).get_params(test, path)
```

Comment:
Ditto for this except for the `path` default.

Reply:
This comment be applied to all `CommandWithParameters` classes.  In retrospect, using a self.path (or something more specific like `self.yaml_param_path`) in the base `get_params()` method would be cleaner.